common-lib/test/cmr/common/test/generics_documentation.clj

```diff
@@ -0,0 +1,88 @@
+
+(deftest get-toc-data
+  (testing "Ensuring that we can extract the necessary components from a toc header")
+  (is (= (gdocs/get-toc-data header-extract-formatted-search "search" (get options-map-search :spacer (fn [x] x))) "    * [content](#link)\n"))
+  (is (= (gdocs/get-toc-data header-extract-formatted-ingest "ingest" (get options-map-ingest :spacer (fn [x] x))) "        * [content](#link)\n")))
+
+(deftest build-markdown-toc-test
+  (testing "Ensuring that the markdown can be created from all of the headers for a given generic
+    retrived from the ingest or search .md")
+  (is (= (gdocs/build-markdown-toc 4 "GridInfo" "grid-link") "    * [grid-link](#GridInfo)\n")))
+
+(def formatted-search-grid-markdown "* [Grid](#grid)\n    * [Searching for grids](#searching-for-grids)\n        * [Grid Search Parameters](#grid-search-params)\n        * [Grid Search Response](#grid-search-response)\n    * [Retrieving All Revisions of a Grid](#retrieving-all-revisions-of-a-grid)\n    * [Sorting Grid Results](#sorting-grid-results)\n")
+(def formatted-ingest-grid-markdown "    * [Grid](#grid)\n            * [/providers/&lt;provider-id&gt;/grids/&lt;native-id&gt;](#provider-info-grid)\n        * [PUT - Create / Update a Grid](#create-update-grid)\n        * [DELETE - Delete a Grid](#delete-grid)\n")
+(def default-formatted-ingest-default "  * [Grid](#grid)\n    * [/providers/&lt;provider-id&gt;/grids/&lt;native-id&gt;](#provider-info-grid)\n   * [PUT - Create / Update a Grid](#create-update-grid)\n   * [DELETE - Delete a Grid](#delete-grid)\n")
+
+(deftest format-generic-toc-test
+  (testing "Ensuring that the generic-toc returns a formatted markdown")
+  (is (= formatted-search-grid-markdown (gdocs/format-generic-toc "search" "grid" "0.0.1" options-map-search)))
+  (is (= formatted-ingest-grid-markdown (gdocs/format-generic-toc "ingest" "grid" "0.0.1" options-map-ingest)))
+  ;;Coverge if the function is passed with a map with the spacer function
+  (is (= default-formatted-ingest-default (gdocs/format-generic-toc "ingest" "grid" "0.0.1" {})))
+  (is (= "" (gdocs/format-generic-toc "none" "grid" "0.0.1" options-map-ingest))))
+
+(deftest all-generic-docs-toc-test
+  (testing "Ensuring that the generic-toc returns a formatted markdown")
+  (is (= true (string/includes? (gdocs/all-generic-docs-toc "search" options-map-search) "[Grid](#grid)"))))
+
+(def buried-html-item "<ul><li>Grid html</li></ul>")
+(def buried-html-item-newline "<ul> \n <li>Grid html</li></ul>")
+
+(deftest format-toc-into-doc-test
+  (testing "Due to preexisting formatting in the api.md documents we need to
+  remove the outer list item and unordered list item tags to ensure that the generics,
+table of contents does NOT break formatting on the documentation")
+  (is (= "Grid html" (gdocs/format-toc-into-doc buried-html-item)))
+  (is (= "Grid html" (gdocs/format-toc-into-doc buried-html-item-newline))))
+
+(def map1 {:a 1 :b 2})
```

Comment:
Delete everything from here down.  While it is good for your learning on how intersection works, we don't need to write tests for clojure functions.